src/translate.rs

```diff
@@ -184,6 +184,40 @@ pub struct StashMut<'a, P: Copy, T: ?Sized>(pub P, pub <T as ToGlibPtrMut<'a, P>
 where
     T: ToGlibPtrMut<'a, P>;
 
+#[derive(Debug)]
+pub struct Borrowed<T>(Option<T>);
```

Comment:
```suggestion
pub struct Borrowed<T>(mem::ManuallyDrop<T>);
```

This is better, it does not involve the annoying `Option` wrapping

Reply:
What will be with `Borrowed::new(None)` for `Option` ?

Reply:
I don't understand :) In that case `T` is `Option<_>` and all is good

Reply:
Really, I missed that :(